share/eds/store_test.go

```diff
@@ -156,6 +156,33 @@ func TestEDSStore_Has(t *testing.T) {
 	assert.True(t, ok)
 }
 
+// TestEDSStore_GC verifies that unused transient shards are collected by the GC periodically.
+func TestEDSStore_GC(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	gcInterval = time.Second
+	edsStore, err := newStore(t)
+	require.NoError(t, err)
+	// kicks off the gc goroutine
+	err = edsStore.Start(ctx)
+	require.NoError(t, err)
+
+	eds, dah := randomEDS(t)
+
+	err = edsStore.Put(ctx, dah, eds)
+	assert.NoError(t, err)
+
+	// doesn't exist yet
+	assert.NotContains(t, edsStore.lastGCResult.Shards, shard.KeyFromString(dah.String()))
+
+	time.Sleep(time.Second)
+	assert.Contains(t, edsStore.lastGCResult.Shards, shard.KeyFromString(dah.String()))
```

Comment:
Could this potentially be flaky in CI? If there is no signals available from Maybe we can run for loop with retry for limited amount time

Reply:
Agree, use retry please

Reply:
done